modules/libcom/src/osi/osiSock.h

```diff
@@ -28,185 +34,367 @@ struct sockaddr;
+ * \return A new socket used for communicating with the peer just accepted, or -1 on error.
+ */
 LIBCOM_API int epicsStdCall epicsSocketAccept ( 
     int sock, struct sockaddr * pAddr, osiSocklen_t * addrlen );
+/*!
+ * \brief Close and free resources held by a SOCKET object.
+ *
+ * Close and free resources held by a SOCKET object.
+ */
 LIBCOM_API void epicsStdCall epicsSocketDestroy ( 
     SOCKET );
+/*!
+ * \brief Allows a socket to bind ignoring other sockets in TIME_WAIT state.
+ *
+ * Allows a socket to bind while other sockets hold the same address and port
+ * but are in the TIME_WAIT state.  TIME_WAIT is a state where a socket has
+ * closed locally, but is waiting to be closed on the remote end.  If the remote
+ * end doesn't respond, this can block another socket from binding to that
+ * address:port for a long time (possibly minutes).  This option allows the
+ * socket to bind to an address:port combo, even if there's another socket with
+ * the same address:port that's waiting for the remote side to close its
+ * connection.
+ *
+ * See also: Linux's SO_REUSEADDR in setsockopt.
+ *
+ * \param s The socket to ignore other TIME_WAIT sockets when binding
+ */
 LIBCOM_API void epicsStdCall 
     epicsSocketEnableAddressReuseDuringTimeWaitState ( SOCKET s );
+/*!
+ * \brief Allows multiple sockets to use the same family, local address, and local
+ * port
+ *
+ * Allows multiple sockets to use the same family, local address, and port.  By
+ * having multiple sockets open with this setting enabled, the OS will
+ * distribute the datagrams (for UDP sockets) or connection requests (for
+ * TCP/stream sockets) among the sockets with this option set.  Note that all
+ * sockets binding to the same address and port must have this option set.
+ *
+ * As an example, this option could be a way to implement a form of load
+ * balancing between multiple threads or processes with sockets bound to the
+ * same address:port.
+ *
+ * \param s The socket to put in "Fanout" mode (aka REUSEPORT mode)
+ */
 LIBCOM_API void epicsStdCall 
     epicsSocketEnableAddressUseForDatagramFanout ( SOCKET s );
 
-/*
- * Fortunately, on most systems the combination of a shutdown of both
- * directions and or a signal is sufficent to interrupt a blocking send,
- * receive, or connect call. For odd ball systems this is stubbed out in the
- * osi area.
+/*!
+ * \brief Enum specifying how to interrupt a blocking socket system call
+ *
+ * This enum specifies how to interrupt a blocking socket system call.
+ * Fortunately, on most systems the combination of a shutdown of both directions
+ * and/or a signal is sufficent to interrupt a blocking send, receive, or
+ * connect call. For odd ball systems this is stubbed out in the osi area.
  */
 enum epicsSocketSystemCallInterruptMechanismQueryInfo {
     esscimqi_socketCloseRequired,
     esscimqi_socketBothShutdownRequired,
     esscimqi_socketSigAlarmRequired /* NO LONGER USED/SUPPORTED */
```

Comment:
I think it would be nice to add a doc comment above saying this variant is deprecated

Reply:
Good idea, I've done that now.